### ISSUE TITLE
The type javax.annotation.Nonnull cannot be resolved

### ISSUE THREAD

**Issue by yotov** [redhat-developer/vscode-java#2712]
In one of my projects, I'm getting an error after upgrading to the latest version:

```
The type javax.annotation.Nonnull cannot be resolved. It is indirectly referenced from required .class filesJava(16777540)
Annotation type 'javax.annotation.Nonnull' cannot be found on the build path, which is implicitly needed for null analysisJava(536871894)
```

I saw the changelog and found that the issue relates to _enhancement - Enable annotation-based null analysis. See [#1693](https://github.com/redhat-developer/vscode-java/issues/1693)._

Removing "javax.annotation.Nonnull" from "java.compile.nullAnalysis.nonnull" config fixes the problem for me.

##### Environment
- Operating System: MacOS 12.6
- JDK version: openjdk 17 2021-09-14
- Visual Studio Code version: 1.71.2
- Java extension version: 1.11.0

**Comment by rgrunber** [comment]
Do you have a sample project available to share that was causing this issue. Looks like the annotations are used in some library/class files but aren't on the classpath ?

CC @CsCherrYY

**Comment by Marinell0** [comment]
I have the same problem now on my project, but a different error: 

Annotation type 'org.springframework.lang.NonNull' cannot be found on the build path, which is implicitly needed for null analysis (Java 536871894)

Sometimes it shows up at the first line, sometimes at a lambda.

**Comment by rgrunber** [comment]
If anyone is able to point to a sample project (or upload one) that causes this problem, it would be very helpful.

**Comment by chowethan** [comment]
This is also occurring on a project I'm a part of. It's preventing the project from building within the IDE. Unfortunately, I'm not able to share any source code.

Is there a way to turn off/disable null analysis in configuration for the time being?

**Comment by nam178** [comment]
I'm having the same problem.
I'm not using `javax.annotation.Nonnull` anywhere in the code other than lombok's `NonNull`
Downgraded Java extension  to `1.10.0` + multiple `mvn clean install` fixed it

**Comment by rgrunber** [comment]
You should be able to disable this feature with the following settings (on the individual workspace, or globally).

**.vscode/settings.json**
```
{
    "java.compile.nullAnalysis.nonnull": [
    ],
    "java.compile.nullAnalysis.nullable": [
    ]
}
```

**Comment by snjeza** [comment]
A related issue - https://github.com/redhat-developer/vscode-java/issues/2721

**Comment by rgrunber** [comment]
@yotov thank-you. Am able to reproduce.

Looking at the classpath of the project, I see both `com.google.code.findbugs:jsr305:jar:3.0.2:runtime` & `org.springframework:spring-core:jar:5.3.22:compile`. So basically we have 2 annotation providers on the classpath. It looks like the defaults prefer the `jsr305` provides (eg. javax.annotation..) over the spring framework provider (eg. org.springframework.lang..). However clearly in this case it should be reversed.

Can you try, the following setting ? It's basically just the default settings, but the spring framework annotations are moved to the top.

**.vscode/settings.json**
```json
{
	"java.compile.nullAnalysis.nullable": [
		"org.springframework.lang.Nullable",
		"javax.annotation.Nullable",
		"org.eclipse.jdt.annotation.Nullable"
	],
	"java.compile.nullAnalysis.nonnull": [
		"org.springframework.lang.NonNull",
		"javax.annotation.Nonnull",
		"org.eclipse.jdt.annotation.NonNull"
	]
}
```

**Comment by yotov** [comment]
@rgrunber yes, reordering the classes fixes the problem and I'm able to compile the project from IDE

**Comment by snjeza** [comment]
> At https://github.com/eclipse/eclipse.jdt.ls/blob/f8452b422add64f9bbc80d0e22b48ed71797e563/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java#L2055 , could we just set the classpath options scope to be "compile" ?

We can't because the scope is set in upstream project (spring-core). m2e adds the classpath test attribute for the maven runtime and test scope.
The PR (https://github.com/eclipse/eclipse.jdt.ls/pull/2264) skips all test classpath entries. It enables null analysis only if there is an annotation type in a classpath entry without the test attribute.
For instance, the vscode-nullable project
This PR enables null analysis for org.springframework.lang.* because the org.springframework:spring-core classpath entry has no the test attribute. The PR skips the jsr305 classpath entry because it has the test attribute. You shouldn't reorder java.compile.nullAnalysis.*.

**Comment by chowethan** [comment]
> @yotov Could you try to remove .project, .classpath, .settings from your project?

@snjeza Mentioned the wrong person, but combined with @yotov's settings options, this worked to fully disable null analysis in my case

**Comment by jvansant** [comment]
> > @yotov Could you try to remove .project, .classpath, .settings from your project?
> 
> @snjeza Mentioned the wrong person, but combined with @yotov's settings options, this worked to fully disable null analysis in my case

It took adding this to my user and workspace settings.json

**Comment by CsCherrYY** [comment]
@jvansant The fix https://github.com/eclipse/eclipse.jdt.ls/pull/2264 is available in the newest verison `1.12.2022101204`, you can try to switch to this version to see if it works.
